### ISSUE TITLE
SSL connection when proxy is no SSL server

### ISSUE THREAD

**Issue by jrgkraus** [larshp/abapGit#2959]
I have problems to authenticate on the proxy server. ZABAPGIT_TEST_SSL says:

IcmConnInitClientSSL: Proxy connection to https://github.com:443 via GCPROXY1.CORP.GEZE.COM:8080 failed (proxy returned 407 authenticationrequired)

Debugging it, I found out that the credentials are being encrypted before authenticating on the proxy. This encryption seems to be forced by the https prefix of the destination url. 

Now, the proxy wants unencrypted credentials, while github needs encryption. Is there a way to resolve this conflict?

**Comment by larshp** [comment]
hmm, SSL is always encrypted and happens before sending the headers(authentication)

I assume you have set the proxy username and password in the settings?

Does it fail at this step in the code?
https://github.com/larshp/abapGit/blob/5018eae46cbc559ede07ba7ef6ce1569ca691a65/src/http/zcl_abapgit_http.clas.abap#L172

**Comment by flaiker** [comment]
Not sure if related but I had a similiar situation where the issue was exactly at the encryption of the proxy user and password. The proxy server retrieved different logon data than the one I specified. I still cannot specify the proxy authentication data in SOAMANAGER at all on service consumer level on some systems. I was able to solve that problem by specifying the parameters globally for the whole client in SICF -> Client -> Proxy Settings and emptying all other settings in SOAMANAGER. I assume it's due to some kind of version mismatch between BASIS, kernel and cryptolib.
